src/server_manager/web_app/ui_components/outline-server-view.js

```diff
@@ -801,43 +803,61 @@ export class ServerView extends DirMixin(PolymerElement) {
     }
   }
 
-  _formatBytesTransferred(numBytes, emptyValue = '') {
+  _makeReadableDataAmount(amount, numDecimals) {
+    const readable = byte_size(amount, {precision: numDecimals});
+    return {
+      amount: readable.value,
+      // Make the unit singular
+      unit: readable.long.slice(0, -1)
+    }
+  }
+
+  _getFormatPrecision(numBytes) {
+    if (numBytes >= 10 ** 9) {
+      return 2;
+    } else if (numBytes >= 10 ** 6) {
+      return 1;
+    }
+    return 0;
+  }
+
+  _formatBytesTransferred(numBytes, language, emptyValue = '') {
```

Comment:
Let's get rid of _formatBytesTransferred and _getFormattedTransferredUnit and:

- Create properties for the value and the unit to use in the HTML
- Add a setter for `totalInboundBytes` and remove setServerTransferredData. The setter should set the two new properties.

The two new properties should be set by extracting from the parts returned by the formatter.

This way you format the data only once and we can see all the logic in one place. It's a bit hard to follow now.

Reply:
I didn't completely get rid of _formatBytesTransferred, as it also provides default values in a number of places.  I did reduce use of it and have it call into the i18n library.  79fb1ea